usefulUtilities/hiFiCalendar/entityServerScripts/tokenServer.js

```diff
@@ -0,0 +1,254 @@
+        "tokenCheck"
+    ];
+
+    // This function is the listener for messages on the message mixer related to this app.
+    this.messageHandler = function(channel, message, senderUUID, localOnly) {
+        if (channel !== CHANNEL) {
+            return;
+        } else {
+            try {
+                message = JSON.parse(message);
+            } catch (e) {
+                console.log(e, "Could not parse message");
+                return;
+            }
+            if (message.type === "APP STARTED") {
+                that.enteredDomain(that.entityID);
+            }
+        }
+    };
+
+
+    this.preload = function(entityID) {
+        that = this;
+        that.entityID = entityID;
+
+        that.request = Script.require('https://hifi-content.s3.amazonaws.com/Experiences/Releases/modules/request/v1.0/request.js').request;
+
+        that.tokenStatus = false;
+        that.refreshCount = 0;
+        that.roomConfigured = false;
+
+        that.entityProperties = Entities.getEntityProperties(that.entityID, ['userData', 'name', 'privateUserData']);
+
+        if (that.entityProperties.userData.length !== 0) {
+            try {
+                that.userData = JSON.parse(that.entityProperties.userData);
+                that.calendarScheduleIDs = that.userData.calendarScheduleIDs;
+                that.privateUserData = that.entityProperties.privateUserData;
+                that.timezoneOffset = that.userData.timezoneOffset;
+                that.timezone = that.userData.timezone;
+                if (that.privateUserData.length > 0) {
+                    that.privateUserData = JSON.parse(that.entityProperties.privateUserData);
+                    that.token = that.privateUserData.token;
+                    that.refreshToken = that.privateUserData.refreshToken;
+                }
+                if (that.userData.roomConfigured) {
+                    that.roomConfigured = that.userData.roomConfigured;
+                }
+                if (that.userData.roomConfig) {
+                    that.roomConfig = that.userData.roomConfig;
+                    that.makeCalendarToRoomAddressMap(that.roomConfig);
+                }
+                if ((new Date().valueOf() + that.userData.timezoneOffset * MS_TO_SEC * SEC_PER_MIN * MIN_PER_HR) > that.userData.expireTime) {
+                    that.calendarScheduleIDs.forEach(function(calendar){
+                        that.tokenCheck(that.entityID, [that.privateUserData.token, calendar.id]);
+                    });
+                }
+            } catch (e) {
+                console.log(e, "Could not parse userData");
+                return;
+            }
+        } else {
+            console.log("Please enter appropriate userData to enable functionality of this server script.");
+            return;
+        }
+        
+        Script.setTimeout(function(){
+            Script.loadEntityScript(that.entityID, Script.resolvePath("./") + SCRIPT_NAME, false);
```

Comment:
I kept this in here in case something unexpected happened in the domain where this wouldn't work anymore, but refreshing the script would always bring it back.  I'm tempted to take it out now that everything seems to be a bit more stable and that might have been helping sidestep some of the initial brittle code.  To your other question, it just seems to refresh the server script that is there.  I think I'm ok with taking these lines out and seeing if the smoke tests start failing for any reason since they have been at 100% for the last couple of weeks.